src/util/transition/ngbTransition.spec.ts

```diff
@@ -337,14 +337,75 @@ if (isBrowserVisible('ngbRunTransition')) {
       expect(element.classList.contains('ngb-test-after')).toBe(false);
     });
   });
+
+  describe('ngbRunTransition nesting', () => {
+
+    beforeEach(() => TestBed.configureTestingModule({declarations: [TestComponentNested]}));
+
+    it(`should ignore all inner element transitions`, (done) => {
+      const fixture = TestBed.createComponent(TestComponentNested);
+      fixture.detectChanges();
+
+      const outerEl = fixture.componentInstance.outer.nativeElement;
+      const innerEl = fixture.componentInstance.inner.nativeElement;
+
+      const nextSpy = createSpy();
+      const errorSpy = createSpy();
+
+      spyOn(fixture.componentInstance, 'onTransitionInnerEnd');
+      spyOn(fixture.componentInstance, 'onTransitionOuterEnd');
+
+      ngbRunTransition(outerEl, () => {
+        outerEl.classList.add('ngb-test-hide-outer');
+        innerEl.classList.add('ngb-test-hide-inner');
+      }, {animation: true, runningTransition: 'continue'}).subscribe(nextSpy, errorSpy, () => {
+        expect(fixture.componentInstance.onTransitionOuterEnd).toHaveBeenCalledTimes(2);
+        expect(fixture.componentInstance.onTransitionInnerEnd).toHaveBeenCalledTimes(1);
+        expectOpacity(outerEl, '0');
+        expectOpacity(innerEl, '0');
+        expect(nextSpy).toHaveBeenCalledWith(undefined);
+        expect(errorSpy).not.toHaveBeenCalled();
+        done();
+      });
+    });
+  });
```

Comment:
Wow, we might need a comment here with a schema and explanations. Schema with DOM structure, what node are transitioned, in which order. Because here, I can not extract any sense out of this test like that. I don't understand at all what's going on.

**EDIT**: I know the DOM is below in the component definition. But still, wouldn't hurt to have a description colocated with the test.

Reply:
Yeah, the DOM is there... I mean do you want me to copy DOM as a comment or what? 
In general I agree that these tests are not easy to read... 😕

Reply:
> I mean do you want me to copy DOM as a comment or what?

No!! 😰 but sometimes, test component should be colocated to the test, and not at the end of the file.